projects/storefrontlib/src/cms-structure/page/slot/page-slot.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core';
+import { DOCUMENT, isPlatformBrowser } from '@angular/common';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PageSlotService {
+  protected instantSsrSlots: string[] | undefined;
+
+  constructor(
+    @Inject(PLATFORM_ID) platformId: any,
+    @Inject(DOCUMENT) document
+  ) {
+    if (isPlatformBrowser(platformId)) {
+      // Find all slots visible in the SSR pre-rendered DOM
+      this.instantSsrSlots = Array.from(
+        document.querySelectorAll('cx-page-slot')
+      )
+        .filter(
+          (el: Element) =>
+            el.getBoundingClientRect().top <
+            document.documentElement.clientHeight
+        )
+        .map((el: Element) => el.classList[0]);
```

Comment:
This is a bit tricky. I'd consider introducing a new page slot attribute that indicates the page slot position.

Reply:
👍